common/crypto_test.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/crypto.hpp>
+
+#include <string>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace std;
+
+namespace error = mender::common::error;
+
+namespace mender {
+namespace common {
+namespace crypto {
+
+TEST(CryptoTest, TestSign) {
+	string data_ {"foobar"};
+	vector<uint8_t> testdata {data_.begin(), data_.end()};
+	string private_key_file = "./private-key.pem";
+	auto expected_signature = crypto::Sign(private_key_file, testdata);
+	ASSERT_TRUE(expected_signature) << "Unexpected: " << expected_signature.error();
```

Comment:
Should test one negative as well, or?

Reply:
Ah, only now did I realize that you are not *verifying* the signature, just making it. My idea was to test an invalid signature, but I guess it doesn't make sense here since we don't verify on the client side. All good then.